### PR TITLE
Bunch of updates for package_manager

### PR DESCRIPTION
**Pull request by tresni**
FAO: @kdeldycke

* Add support for pip3, node, atom, ruby gems
* Fixup brew cask checking
* Don't die on errors

### CODE REVIEW

File: Dev/Homebrew/package_manager.7h.py [hunk 1]
```diff
@@ -186,13 +193,117 @@ def update_all_cli(self):
...
+            })
+
+    def update_cli(self, package_name=None):
+        cmd = "{} -g --progress=false update".format(self.cli)
+        if package_name:
+            cmd += " {}".format(package_name)
+        return self.bitbar_cli_format(cmd)
+
+    def update_all_cli(self):
+        return self.update_cli()
```
Comment (keithamus) [inline Dev/Homebrew/package_manager.7h.py]: Dang I was going to do this 😆

File: Dev/Homebrew/package_manager.7h.py [hunk 2]
```diff
@@ -186,13 +193,117 @@ def update_all_cli(self):
...
+
+    @property
+    def name(self):
+        return "npm"
+
+    def sync(self):
+        output = self.run(self.cli, '-g', '--progress=false', '--json',
+                          'outdated')
+        for package, values in json.loads(output).iteritems():
+            self.updates.append({
```
Comment (keithamus) [inline Dev/Homebrew/package_manager.7h.py]: you should add an `if values['current'] != 'linked':` here.

Comment (tresni) [inline Dev/Homebrew/package_manager.7h.py]: Hopefully you fixed that :)  I don't have any linked packages (nor have I ever) so yeah, that's a new one for me :)

**Comment by kdeldycke** [comment]
Wow, you beat me to it! Good job guys! That's the spirit of the plugin! :)

**Comment by kdeldycke** [comment]
Just a little tweak: can you update the version from v1.1 to v1.2?

File: Dev/Homebrew/package_manager.7h.py (hunk 2 above)
Comment (keithamus) [inline Dev/Homebrew/package_manager.7h.py]: Fix coming shortly.

**Pull request merged**